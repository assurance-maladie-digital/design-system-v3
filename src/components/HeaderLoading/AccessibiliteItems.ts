import { ExpertiseLevelEnum } from './constants/ExpertiseLevelEnum'

export const AccessibiliteItemsIndeterminate = [
	
	{
		title: 'Catégorie 10 : Présentation de l’information',
		subtitle: '10.1 Dans le site web, des feuilles de styles sont-elles utilisées pour contrôler la présentation de l’information ? ?',
		items: [

			{
				precision: 'Dans chaque page web, les balises servant à la présentation de l’information ne doivent pas être présentes dans le code source généré des pages. Cette règle est-elle respectée ? ',
				link: 'https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#10.1.1',
				solution: [{
					info1: '1. Vérifier l’absence des éléments de présentation <basefont>, <big>, <blink>, <center>, <font>, <marquee>, <s>, <strike>, <tt> ; ',
					info2: '2. Vérifier l’absence de l’élément <u> uniquement si le DOCTYPE du document ne correspond pas à HTML 5 ; ',
					info3: '3. Si c’est le cas, le test est validé',
				}],
				expertise: ExpertiseLevelEnum.DEV,

			},

		],
	},

]

export const AccessibiliteItemsValidated = [
	
]
