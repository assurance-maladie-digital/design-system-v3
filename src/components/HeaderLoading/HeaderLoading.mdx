import { Controls, Canvas, Meta, Source } from '@storybook/blocks';
import * as HeaderLoading from './HeaderLoading.stories';

<Meta of={HeaderLoading} />

# HeaderLoading

L’élément `HeaderLoading` est une extension du composant `VSkeletonLoader` de Vuetify, il est utilisé pour afficher un élément de chargement avec des dimensions personnalisées.

<Canvas of={HeaderLoading.Default} />

# API

<Controls of={HeaderLoading.Default} />

# Exemple d'utilisation

<Source dark code={`
<script setup lang="ts">
	import HeaderLoading from '@/components/HeaderLoading/HeaderLoading.vue'
</script>

<template>
	<div class="pa-4">
		<HeaderLoading width="200px" />
	</div>
</template>
`} />