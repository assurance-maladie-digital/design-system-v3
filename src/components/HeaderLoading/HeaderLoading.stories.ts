import HeaderLoading from './HeaderLoading.vue'
import type { Meta, StoryObj } from '@storybook/vue3'

const meta: Meta<typeof HeaderLoading> = {
	title: 'Composants/Feedback/HeaderLoading',
	component: HeaderLoading,
	parameters: {
		layout: 'fullscreen',
		controls: { exclude: ['heading', 'itemsNumber', 'row'] },
	},
	argTypes: {
		width: { control: 'text' },
		height: { control: 'text' },
	},
}

export default meta

type Story = StoryObj<typeof meta>

export const Default: Story = {
	parameters: {
		sourceCode: [
			{
				name: 'Template',
				code: `
				<template>
					<HeaderLoading :width="width" :height="height" />
				</template>
				`,
			},
			{
				name: 'Script',
				code: `
				<script setup lang="ts">
					import HeaderLoading from '@cnamts/synapse'
					
					const width = '100px'
					const height = '1rem'
				</script>
				`,
			},
		],
	},
	args: {
		width: '100px',
		height: '1rem',
	},
	render: (args) => {
		return {
			components: { HeaderLoading },
			setup() {
				return { args }
			},
			template: `
				<div class="pa-4">
                    <HeaderLoading v-bind="args" />
				</div>
			`,
		}
	},
}
