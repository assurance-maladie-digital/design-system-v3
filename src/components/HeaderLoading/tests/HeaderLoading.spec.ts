import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import { vuetify } from '@tests/unit/setup'

import HeaderLoading from '../HeaderLoading.vue'

describe('HeaderLoading', () => {
	it('renders correctly', () => {
		const wrapper = mount(HeaderLoading, {
			global: {
				plugins: [vuetify],
			},
			propsData: {
				width: '100px',
				height: '1rem',
			},
		})

		expect(wrapper.html()).toMatchSnapshot()
	})
})
