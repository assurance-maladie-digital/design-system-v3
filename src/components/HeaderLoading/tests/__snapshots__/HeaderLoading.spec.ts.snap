// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeaderLoading > renders correctly 1`] = `
"<div data-v-947c5508="" class="v-skeleton-loader v-theme--light vd-header-loading" style="height: 1rem; width: 100px;" arialive="polite" arialabel="Loading..." role="alert" aria-hidden="true">
  <div class="v-skeleton-loader__bone v-skeleton-loader__heading"></div>
</div>"
`;
